post_office/validators.py

```diff
@@ -1,28 +1,31 @@
+import re
+
 from django.core.exceptions import ValidationError
-from django.core.validators import validate_email
+from django.core.validators import validate_email as validate_plain_email, EmailValidator
 from django.template import Template, TemplateSyntaxError
 from django.utils.encoding import force_text
 
 from .compat import text_type
 
-
-def validate_email_with_name(value):
-    """
-    Validate email address.
+class FullEmailValidator(EmailValidator):
+    """ Simple validator that passes for email addresses bearing a display name
+    i.e. John Smith <john.smith@acme.gov>
 
     Both "Recipient Name <email@example.com>" and "email@example.com" are valid.
-    """
-    value = force_text(value)
 
-    if '<' and '>' in value:
-        start = value.find('<') + 1
-        end = value.find('>')
-        if start < end:
-            recipient = value[start:end]
-    else:
-        recipient = value
+    """
+    def __call__(self, value):
+        try:
+            res = super(FullEmailValidator, self).__call__(value)
```

Comment:
I wonder if it's more efficient to detect whether display name is present before performing the super call. This way, we can avoid calling Django's builtin email validation twice in the event that email with name is passed in as an argument. Django's builtin validator performs a few regex matches which are relatively expensive compared to simple string based checks.

I don't think we can answer this with no real world usage data though :\